src/test/java/com/toofifty/easyblastfurnace/EasyBlastFurnacePluginTest.java

```diff
@@ -1,13 +1,364 @@
 package com.toofifty.easyblastfurnace;
```

Comment:
Still can't believe you added tests! I'm more than happy to pull these in as-is :) 

My suggestion though would be to split them up a bit more and make them a bit _dumber_ - having if statements in tests usually means your tests are doing too much. Same with having assertions extracted outside of the test method body IMO. Though extraction of _setup_ tasks is definitely helpful for readability.

Here's how I think I'd structure one of these:

```java
public void mithrilHybridMethod()
{
    setInventoryCount(ItemID.MITHRIL_ORE, 1);
    setInventoryCount(ItemID.GOLD_ORE, 1);
    assertEquals(Strings.MITHRIL_BAR_METHOD, methodHandler.getMethod().getName());

    assertStepTooltip(Strings.WITHDRAW_COAL_BAG);

    setInventoryCount(ItemID.COAL_BAG, 1);
    // or even something like withdrawCoalBag();
    assertStepTooltip(Strings.FILL_COAL_BAG);

    fillCoalBag();
    assertStepTooltip(Strings.WITHDRAW_ICE_OR_SMITHS_GLOVES);

    setInventoryCount(ItemID.ICE_GLOVES, 1);
    setBankCount(ItemID.SMITHING_CAPE, 1);
    assertStepTooltip(Strings.WITHDRAW_SMITHING_CAPE);

    // ...
}

Reply:
Okay, I'll do my best to make it look more like that

Reply:
@Toofifty I've done my best to do as you suggested. Please take a look and let me know if you want me to do anything else.